src/script/cpp_api/s_security.cpp

```diff
@@ -236,6 +236,157 @@ void ScriptApiSecurity::initializeSecurity()
 	lua_pop(L, 1); // Pop globals_backup
 }
 
+void ScriptApiSecurity::initializeSecurityClient()
+{
+	static const char *whitelist[] = {
+		"assert",
+		"core",
+		"collectgarbage",
+		"DIR_DELIM",
+		"error",
+		"getfenv",
+		"ipairs",
+		"next",
+		"pairs",
+		"pcall",
+		"print",
+		"rawequal",
+		"rawget",
+		"rawset",
+		"select",
+		"setfenv",
+		"setmetatable",
+		"tonumber",
+		"tostring",
+		"type",
+		"unpack",
+		"_VERSION",
+		"xpcall",
+		// Completely safe libraries
+		"coroutine",
+		"string",
+		"table",
+		"math",
+	};
+	static const char *io_whitelist[] = {
+		"close",
```

Comment:
I agree to rubenwardy. Deciding which path to access is quite hard if you consider the corner cases. If you limit to a single mod specific folder you're basically imitating a virtual filesystem. If not you risk missing some critical parts. e.g. /tmp usually is fine for local applications but there's a lot of data stored in /tmp you don't want someone from outside your machine to access.